src/main/java/backend/Logic.java

```diff
@@ -264,9 +265,20 @@ public Model getRepo(String repoId) {
                 repoOpControl.replaceIssueLabelsLocally(issue, newLabels);
         localLabelsReplaceFuture.thenRun(this::refreshUI);
 
-        return updateIssueLabelsOnServer(issue, newLabels)
+        updateIssueLabelsOnServer(issue, newLabels)
+                .exceptionally((e) -> {
+                    result.completeExceptionally(e);
+                    return false;
+                })
                 .thenCombine(localLabelsReplaceFuture, (isUpdateSuccessful, locallyModifiedIssue) ->
-                        handleIssueLabelsUpdateResult(isUpdateSuccessful, locallyModifiedIssue, originalLabels));
+                        handleIssueLabelsUpdateResult(isUpdateSuccessful, locallyModifiedIssue, originalLabels))
+                .thenAccept((isReplacementSuccessful) -> {
+                    if (!result.isCompletedExceptionally()) {
```

Comment:
This part is a bit confusing -- we propagate failure using `completeExceptionally` above as well as with a boolean value. It's not clear what the difference between the two is from just reading this part. Is there a way to simplify it, like maybe `completeExceptionally` only at the end if `isReplacementSuccessful` is false?